.github/scripts/fetch_latest_green_commit.py

```diff
@@ -0,0 +1,156 @@
+import json
+import os
+import re
+import sys
+from typing import Any, cast, Dict, List, NamedTuple, Tuple
+
+import rockset  # type: ignore[import]
+from gitutils import _check_output
+
+
+def eprint(msg: str) -> None:
+    print(msg, file=sys.stderr)
+
+
+class WorkflowCheck(NamedTuple):
+    workflowName: str
+    name: str
+    jobName: str
+    conclusion: str
+
+
+def get_latest_commits() -> List[str]:
```

Comment:
Why return a list? Shouldn't be one commit?

Reply:
For this, the scripts works by returning a list of commits between `viable/strict` and main here.  Then it will check those commits one by one starting from the latest main and return the first commit it finds where all required jobs are passing.  This will become the candidate commit for `viable/strict`.